### PR TITLE
feat: doublezero template

### PR DESCRIPTION
**Pull request by ciocan**
Added [DoubleZero](https://github.com/technomancy-dev/00/) template

### CODE REVIEW

File: templates/doublezero/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+services:
+  doublezero:
+    restart: always
+    image: liltechnomancer/double-zero:latest
```
Comment (Siumauricio) [inline templates/doublezero/docker-compose.yml]: use a specific version not latest

File: templates/doublezero/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    ports:
+      - 4000:4000
```
Comment (Siumauricio) [inline templates/doublezero/docker-compose.yml]: Don't expose the port instead do this:

```yaml
ports:
    - ${DOUBLEZERO_PORT}
```

File: templates/doublezero/index.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+} from "../utils";
+
+export function generate(schema: Schema): Template {
+	const mainServiceHash = generateHash(schema.projectName);
+	const randomDomain = generateRandomDomain(schema);
+	const envs = [
+		`DOUBLEZERO_HOST=${randomDomain}`,
+		"DOUBLEZERO_PORT=4000",
+		`HASH=${mainServiceHash}`,
+		"SECRET_KEY_BASE=",
```
Comment (Siumauricio) [inline templates/doublezero/index.ts]: Looking at the documentation it needs a base64, so please use the `generateBase64` function

File: templates/templates.ts [hunk 4]
```diff
@@ -331,5 +331,19 @@ export const templates: TemplateData[] = [
 		},
 		tags: ['chat'],
 		load: () => import('./open-webui/index').then((m) => m.generate),
+	},
+	{
+		id: 'doublezero',
+		name: 'Double Zero',
+		version: 'v0.0',
```
Comment (Siumauricio) [inline templates/templates.ts]: Set the same version as in the service of the compose file

File: templates/doublezero/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      SYSTEM_EMAIL: ${SYSTEM_EMAIL}
+      SECRET_KEY_BASE: ${SECRET_KEY_BASE}
+      PHX_HOST: ${DOUBLEZERO_HOST}
+      DATABASE_PATH: ./00.db
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.${HASH}.rule=Host(`${DOUBLEZERO_HOST}`)"
+      - "traefik.http.routers.${HASH}.entrypoints=websecure"
+      - "traefik.http.routers.${HASH}.tls.certResolver=letsencrypt"
+      - "traefik.http.services.${HASH}.loadbalancer.server.port=${DOUBLEZERO_PORT}"
```
Comment (Siumauricio) [inline templates/doublezero/docker-compose.yml]: remove this lines, when using traefik.me domains is not necessary to use these lines

```suggestion
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.${HASH}.rule=Host(`${DOUBLEZERO_HOST}`)"
      - "traefik.http.services.${HASH}.loadbalancer.server.port=${DOUBLEZERO_PORT}"
```

**Review by Siumauricio — APPROVED** [review]

**Review by Siumauricio — CHANGES_REQUESTED** [review]
Looks really cool template, thank you!

File: templates/templates.ts (hunk 4 above)
Comment (ciocan) [reply templates/templates.ts:338]: they don't have yet a version name, that's why I set 0

**Review by Siumauricio — APPROVED** [review]
Thank you! ❤️

**Pull request merged**